apypie/api.py

```diff
@@ -60,7 +62,35 @@ def __init__(self, **kwargs):
         if kwargs.get('username') and kwargs.get('password'):
             self._session.auth = (kwargs['username'], kwargs['password'])
 
-        self.apidoc = self._load_apidoc()
+        self._apidoc = self._load_apidoc()
+
+    @property
+    def apidoc(self):
+        if self._apidoc is None:
+            self._apidoc = self._load_apidoc()
+        return self._apidoc
+
+    @property
+    def apidoc_cache_file(self):
+        return os.path.join(self.apidoc_cache_dir, '{0}{1}'.format(self.apidoc_cache_name, self.cache_extension))
+
+    def _set_default_name(self, default='default'):
+        """find the newest file in the cachedir if any"""
+        cache_file = sorted(glob.iglob(os.path.join(self.apidoc_cache_dir, '*{}'.format(self.cache_extension))), key=os.path.getctime, reverse=True)
```

Comment:
there will be ever at most one file in that folder, so there is no need to find the newest, next(iglob) is sufficient